lib/src/signed_video_h26x_common.c

```diff
@@ -867,13 +922,19 @@ check_and_copy_hash_to_hash_list(signed_video_t *self, const uint8_t *hash, size
 
   uint8_t *hash_list = &self->gop_info->hash_list[0];
   int *list_idx = &self->gop_info->list_idx;
-  // Check if there is room for another hash in the |hash_list|.
-  if (*list_idx + hash_size > self->gop_info->hash_list_size) *list_idx = -1;
+
+  if (*list_idx + hash_size > self->gop_info->hash_list_size + 1) {
```

Comment:
If `hash_size = 32`, `self->gop_info->hash_list` is declared as `uint8_t hash_list[159];`, `self->gop_info->hash_list_size = 159` and `self->gop_info->list_idx = 128`
what will happen?

Reply:
if hash_size = 32, self->gop_info->hash_list_size = 159, and self->gop_info->list_idx = 128, the "*list_idx + hash_size > self->gop_info->hash_list_size + 1" returns  to false. Therefore, *list_idx is not set to -1 and  hash  can be  put  to last slot of hashlist

Reply:
But then you will write 32 bytes of data when only 159 - 128 = 31 bytes are available, hence write outside memory.